### PR TITLE
Adds ObjectResult for Forbid Response

### LINKED ISSUE #8219: ForbidObjectResult
#### Feature request

It seems useful to have `ForbidObjectResult` as a complement to `ForbidResult` (similar to `NotFoundObjectResult`/`NotFoundResult` and `ConflictObjectResult`/`ConflictResult`).

RFC 7231, [in section 6.5.3.  403 Forbidden](https://tools.ietf.org/html/rfc7231#section-6.5.3), specifically states:

>A server that wishes to make public why the request has been forbidden can describe that reason in the response payload (if any).

### PR DESCRIPTION
**Pull request by Matthiee**
[RFC 7231, in section 6.5.3. 403 Forbidden](https://tools.ietf.org/html/rfc7231#page-59), specifically states:

> A server that wishes to make public why the request has been forbidden can describe that reason in the response payload (if any).

Fixes #8219

### CODE REVIEW

**Comment by pranavkm** [comment]
@Matthiee please see the previous PR for what needs to be done to address this correctly - https://github.com/aspnet/Mvc/pull/8264

**Comment by Matthiee** [comment]
@pranavkm Does `ForbidAsync` need to be called when overriding `ExecuteResultAsync`?

**Comment by pranavkm** [comment]
>  Does ForbidAsync need to be called when overriding ExecuteResultAsync?

Yup Plus both the newly added ActionResult and the action need to document the execution of the action result depends on the implementation of the auth provider. For instance, using this action result would not work correctly with cookie auth.

File: src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+        /// </summary>
+        public IList<string> AuthenticationSchemes { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="AuthenticationProperties"/> used to perform the authentication challenge.
+        /// </summary>
+        public AuthenticationProperties Properties { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs]: This no longer writes the object to the response.

File: src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var loggerFactory = context.HttpContext.RequestServices.GetRequiredService<ILoggerFactory>();
+            var logger = loggerFactory.CreateLogger<ForbidResult>();
+
+            logger.ForbidResultExecuting(AuthenticationSchemes);
+
+            if (AuthenticationSchemes != null && AuthenticationSchemes.Count > 0)
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs]: The pattern for doing this is to invoke an executor - see https://github.com/aspnet/Mvc/blob/release/2.2/src/Microsoft.AspNetCore.Mvc.Core/ObjectResult.cs#L36

File: src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+
+        /// <inheritdoc />
+        public override async Task ExecuteResultAsync(ActionContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var executor = context.HttpContext.RequestServices.GetRequiredService<IActionResultExecutor<ForbidObjectResult>>();
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Mvc.Core/ForbidObjectResult.cs]: This wouldn't work - there isn't an executor registered for this type. Also you'd want the body of this code to live inside the executor.

**Comment by pranavkm** [comment]
Could you send a PR with unit tests and functional tests that verify this works end to end?

**Comment by mkArtakMSFT** [comment]
Thanks for your effort, @Matthiee.
This PR had spun up a conversation on our end and we've realized that we won't actually do anything about the related issue. The reason is described in the issue.

**Pull request closed without merging**